### ISSUE TITLE
`lsl`'s third argument on aarch64 decoded incorrectly

### ISSUE THREAD

**Issue by enedil** [Vector35/binaryninja-api#1870]
Binary Ninja Version: 2.1.2263 Personal, 4d2bf25a
Platform: Fedora 32 (Workstation Edition)

That's what `aarch64-linux-gnu-objdump` says:
```
   10890:	53092108 	lsl	w8, w8, #23
```

And that's the output from Binary Ninja
```
00010890  08210953   lsl     w8, w8, #0xfffffff7
```

**Comment by verylazyguy** [comment]
I just ran into the same thing. The LLIL for mine is `x1 = x9 << -0x3d`

**Comment by enedil** [comment]
Just to add the comment, LLIL decodes this instruction as `w8 = w8 << -9`. As it happens, 32-9 = 23. I suppose this is just a 5 bit encoding, which you somehow complete to 32 bits by sign extension.

**Comment by lwerdna** [comment]
candidate fix in the submodule: https://github.com/Vector35/arm64_disassembler/commit/5134e2d2dae0cc5767fb5e50cf981e48924d9336

will perform more testing tomorrow and update dev's submodule reference and close this if all passes

**Comment by lwerdna** [comment]
The fix should be on dev after the next build: f09be7ef6e2a82a5c4667bf7051459ca6d72f54f
